### ISSUE TITLE
Self Solve URL Changed

### ISSUE THREAD

**Issue by csteelatgburg** [rustymyers/warranty#11]
It looks like Apple has changed the selfsolve.apple.com changed again and the script no longer works.

**Comment by jgstew** [comment]
Seems like this is the issue: http://openradar.appspot.com/radar?id=4998891128750080

Related: https://github.com/chilcote/warranty/issues/6

**Comment by gilbertpalau** [comment]
Crap.  No wonder.... I just ran the script against 600 systems and they all came back and not having any warranty...

I'm guessing it hasn't been fixed by Apple... @rustymyers is there something that could be added to a script to read the captcha or avoid it or its a lost cause?

**Comment by gilbertpalau** [comment]
Actually I  think this is boned. there's a csrf-token in the form and that's generated server side.

**Comment by RobertJansson** [comment]
It is because people tend to send this in a storm from 600 computers at a time :.-/

**Comment by rustymyers** [comment]
Apple has continually removed functions these scripts have utilized. Our org has moved on to using GSX APIs to return warranty status, which provides admins additional data. I'd suggest moving that direction for bulk warranty status operations.

**Comment by csteelatgburg** [comment]
What code are you using to access the GSX API? I'm trying to implement the gsxlib from here https://github.com/filipp/gsxlib and having issues.

**Comment by gilbertpalau** [comment]
except our enterprise doesn’t meet the minimum requirements of 1000 devices
deployed and we can’t get access to it.  That is pretty stupid from Apple’s
part to keep limiting the enterprise support.

On Wed, May 4, 2016 at 9:24 AM, Rusty Myers <notifications@github.com>
wrote:

> Apple has continually removed functions these scripts have utilized. Our
> org has moved on to using GSX APIs to return warranty status, which
> provides admins additional data. I'd suggest moving that direction for bulk
> warranty status operations.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/rustymyers/warranty/issues/11#issuecomment-216881155>
>



-- 
Gilbert J. Palau
gilbertpalau@gmail.com
http://gilbertopalau.com

**Comment by jgstew** [comment]
I agree, this is a bit annoying on Apple's part, but one thing I have done to help mitigate the issue is to automatically create links to the Apple Warranty checker page with the serial number of the computer record you are currenting viewing in our config management system. Then you just have to click on the link, enter the captcha, and get the info. This isn't ideal, but it does make it easier for IT staff that only need to check the warranty status for a few systems in specific circumstances.

This solution doesn't solve lifecycle planning for the fleet of computers as a whole, which is still a problem. Some others have come up with solutions to use the sell date or manufacture date to help estimate the warranty period.

**Comment by hansen-m** [comment]
We've successfully used this Python project from the same developer mentioned: https://github.com/filipp/py-gsxws

With the new certificate and whitelist requirements from Apple it takes a little bit of work to get it setup but we've wrapped it in a very basic Flask app to build our own "self solve" lookup tool, but by using the GSX API it includes much more data than Apple ever provided on their own page. Plus it gives us the added ability to return the data in any format (xml/soap, plist, json) or really customize it anyway you'd like. 

I might be able to make our code available on Github but it would take a little bit of work to cleanup and sanitize.
